lib/open_project/reporting/patches/costlog_controller_patch.rb

```diff
@@ -53,7 +53,9 @@ def index_with_reports_view
 
         if @work_package
           if @work_package.respond_to?("lft")
-            work_package_ids = WorkPackage.all(select: :id, conditions: ["root_id = ? AND lft >= ? AND rgt <= ?", @work_package.root_id, @work_package.lft, @work_package.rgt]).collect{|i| i.id}
+            work_package_ids = WorkPackage
+                               .where(['root_id = ? AND lft >= ? AND rgt <= ?', @work_package.root_id, @work_package.lft, @work_package.rgt])
```

Comment:
This is correct and everything but it also is a complicated way of simply writing
`@work_package.self_and_descendants.pluck(:id)`